xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletResponse.java

```diff
@@ -66,9 +68,21 @@ public void sendRedirect(String redirect) throws IOException
             LOGGER.warn("Possible HTTP Response Splitting attack, attempting to redirect to [{}]", redirect);
             return;
         }
+        if (redirect.matches("[a-z0-9]+://.*")) {
```

Comment:
You should have a pre compiled `Pattern` as constant field for performances.

Reply:
Indeed.